### PR TITLE
docs: update the Output Files Guide page in the Sphinx documentation

### LINKED ISSUE #278: Update the content of the Output Files Guide page in our Sphinx documentation
**Description**
This issue is created to track the progress of updating the content for the `Output Files Guide` page in the Sphinx documentation.
This page contains the description of all the files and directories created by Macaron in the `output` directory.

### PR DESCRIPTION
**Pull request by tromai**
Closes #278

### CODE REVIEW

File: docs/source/pages/output_files.rst [hunk 1]
```diff
@@ -7,4 +7,142 @@
...
+
+    output/
+    └── reports/
+        └── ...
+
+''''''''''''''''''
+Unique result path
+''''''''''''''''''
+
+Macaron creates a different directory under ``reports`` to store the report files for each target repository. This directory
```
Comment (behnazh-w) [inline docs/source/pages/output_files.rst:44]:
```suggestion
For each target repository, Macaron creates a directory under ``reports`` to store the report files. This directory
```

File: docs/source/pages/output_files.rst [hunk 2]
```diff
@@ -7,4 +7,142 @@
...
+path is formed from the git host name (e.g ``github.com``), the owner and the name of that
+repository. The final path is created using the following template:
+
+.. code-block::
+
+    <path_to_output>/reports/<git_service_name>/<owner>/<repo_name>
+
+.. note:: The git host name has all occurrence of ``.`` in the URL replaced by ``_``.
+
+For example, when analyze the repository at `<https://github.com/micronaut-projects/micronaut-core>`_, Macaron will store
```
Comment (behnazh-w) [inline docs/source/pages/output_files.rst:54]:
```suggestion
For example, the reports for  `<https://github.com/micronaut-projects/micronaut-core>`_ repository will be stored under
```

File: docs/source/pages/output_files.rst [hunk 3]
```diff
@@ -7,4 +7,142 @@
...
+the report files into ``<path_to_output>/reports/github_com/micronaut-projects/micronaut-core``.
+
+''''''''''''
+Report types
+''''''''''''
+
+There are currently three types of report that Macaron creates:
```
Comment (behnazh-w) [inline docs/source/pages/output_files.rst:61]:
```suggestion
Macaron creates three types of reports:
```

Comment (tromai) [reply docs/source/pages/output_files.rst]: Addressed in https://github.com/oracle-samples/macaron/pull/281/commits/cacf960a62fb0430209e64eb22e21125a4a57760

File: docs/source/pages/output_files.rst (hunk 2 above)
Comment (tromai) [reply docs/source/pages/output_files.rst]: Addressed in https://github.com/oracle-samples/macaron/pull/281/commits/cacf960a62fb0430209e64eb22e21125a4a57760

File: docs/source/pages/output_files.rst (hunk 1 above)
Comment (tromai) [reply docs/source/pages/output_files.rst]: Addressed in https://github.com/oracle-samples/macaron/pull/281/commits/cacf960a62fb0430209e64eb22e21125a4a57760

**Review by behnazh-w — APPROVED** [review]

**Pull request merged**